include/field2d.hxx

```diff
@@ -150,8 +150,8 @@ class Field2D : public Field, public FieldData {
   const BoutReal& operator[](const Ind2D &d) const {
     return data[d.ind];
   }
-  BoutReal& operator[](const Ind3D &d); 
-  const BoutReal& operator[](const Ind3D &d) const;
+  inline BoutReal& operator[](const Ind3D &d);
```

Comment:
I don't think the compiler can inline a function unless the body is available here in the header.

Reply:
Yes, it has to be "in the same translation unit", so included via this header :(

Reply:
Not only was this not useful, it was actively harmful as it caused the linker to not see the implementation.

Also it turns out it wasn't being tested in the unit tests, so fixed that too